site/app/templates/submission/homework/SubmitBox.twig

```diff
@@ -10,10 +10,57 @@
     <div class="upperinfo">
         <h1 class="upperinfo-left">New submission for: {{ gradeable_name }}</h1>
         {% if has_due_date %}
-            <h3 class="upperinfo-right">Due: {{ formatted_due_date }}</h3>
+            <h3 class="upperinfo-right">Due: {{ formatted_due_date }} <br>
+                {%  if due_in_a_day %}
+                    <span>Now: <span id="current-server-time">{{ formatted_current_time }}</span></span>
+                {% endif %}
+            </h3>
         {% endif %}
     </div>
 
+    {%  if due_in_a_day %}
+    <script>
+        $(function(){
+
+            // Get browser's time, timezone offset and unix timestamp (all in milliseconds)
+            var localDate = new Date();
+            var localTimestamp = localDate.getTime();
+            var localOffset = localDate.getTimezoneOffset() * 60000;
+
+            // Calculate shift to UTC in milliseconds
+            var UTCTime = localTimestamp + localOffset;
+            var serverTimeShift;
+
+            var fetchTimeUrl = window.location.href + "&action=fetch_time";
+
+            setTimeout(function(){
+                $.ajax({
+                    url: fetchTimeUrl,
+                    dataType: "json",
+                    success: function(result){
+                        // timezone offset returned in seconds from server
+                        var serverOffset = result.data.offset * 1000;
+
+                        // server time shift is shift from UTC and the seconds before the ajax call was made is added to it.
+                        serverTimeShift = UTCTime + serverOffset + ((60 - localDate.getSeconds())*1000);
+
+                        setTime(); // Since setInterval runs after the interval and we need an update right now, run the function once
+                        setInterval(function(){setTime()}, 60 * 1000);
```

Comment:
What is the point of starting another setInterval loop here? Wouldn't every asynchronous request create its own loop and you're already launching multiple ajax calls since all of that is within its own loop? 

Also this PR looks similar to #3508 where you mentioned theres a better way of doing this? I'm confused on if you think theres a better way to implement this.

Reply:
@shailpatels there are no multiple ajax calls here, it's just a single fetch at the first 00 seconds on the clock after the page loads, henceforth it is always just calculation of server time by shifting the local time to it, which is being taken from the browser. 

This PR is significantly different from the #3508 because this is the better way about which I talked there. The maximum error there was 5 minutes, here it is 1 minute. 

The function inside the setInterval is the only block being called repeatedly, and not in an asynchronous fashion.

Reply:
The setInterval is not being run more than once!